packages/cloud_firestore/cloud_firestore_web/lib/src/interop/firestore.dart

```diff
@@ -133,17 +144,28 @@ class LoadBundleTask
   ///Tracks progress of loadBundle snapshots as the documents are loaded into cache
   Stream<LoadBundleTaskProgress> stream() {
     late StreamController<LoadBundleTaskProgress> controller;
-    controller =
-        StreamController<LoadBundleTaskProgress>.broadcast(onListen: () {
+    controller = StreamController<LoadBundleTaskProgress>(onListen: () {
       /// Calls underlying onProgress method on a LoadBundleTask [jsObject].
       jsObject.onProgress(
           allowInterop((firestore_interop.LoadBundleTaskProgressJsImpl data) {
-        controller.add(LoadBundleTaskProgress._fromJsObject(data));
+        LoadBundleTaskProgress taskProgress =
+            LoadBundleTaskProgress._fromJsObject(data);
+
+        if (LoadBundleTaskState.error != taskProgress.taskState) {
+          //error handled in addError() call below.
+          controller.add(taskProgress);
+        }
+      }));
+
+      jsObject.then(allowInterop((value) {
+        controller.close();
+      }), allowInterop((error) {
+        controller.addError(FirebaseException(
+            plugin: 'cloud_firestore',
+            message: error.message,
+            stackTrace: StackTrace.fromString(error.stack)));
```

Comment:
Some trailing commas:
```suggestion
            stackTrace: StackTrace.fromString(error.stack),),);
```